### PR TITLE
refactor(exception): separate local raised errors from ReponseError

### PR DESCRIPTION
**Pull request by Lee-000**

### CODE REVIEW

**Comment by GravitiRob** [comment]
Task linked: [CU-2bpma5x 清理请求异常中无用的逻辑](https://app.clickup.com/t/2bpma5x)

**Review by graczhual — APPROVED** [review]

File: graviti/exception.py [hunk 1]
```diff
@@ -49,7 +49,31 @@ def __init__(self, message: Optional[str] = None, *, is_draft: Optional[bool] =
             self._message = f"The status is not {required_status}"
 
 
-class ResponseError(ClientError):
+class OperationForbiddenError(ManagerError):
+    """This class defines the exception for illegal operations that Graviti SDK forbids."""
+
+
+class InvalidResourceNameError(ManagerError):
```
Comment (linjiX) [inline graviti/exception.py:56]: `InvalidResourceNameError` -> `ResourceNameError`?

File: graviti/exception.py [hunk 2]
```diff
@@ -49,7 +49,31 @@ def __init__(self, message: Optional[str] = None, *, is_draft: Optional[bool] =
...
+    """This class defines the exception for invalid resource names."""
+
+    def __init__(
+        self,
+        message: Optional[str] = None,
+        *,
+        resource: Optional[str] = None,
+        identification: Union[int, str, None] = None,
```
Comment (linjiX) [inline graviti/exception.py:64]: should `resource` and `identification` be required parameter?

Comment (linjiX) [reply graviti/exception.py:64]: Also we need to reconsider the variable name of these parameter

File: graviti/exception.py [hunk 3]
```diff
@@ -49,7 +49,31 @@ def __init__(self, message: Optional[str] = None, *, is_draft: Optional[bool] =
             self._message = f"The status is not {required_status}"
 
 
-class ResponseError(ClientError):
+class OperationForbiddenError(ManagerError):
```
Comment (linjiX) [inline graviti/exception.py:52]: The name `OperationForbiddenError` is not clear

**Review by linjiX — CHANGES_REQUESTED** [review]

File: graviti/exception.py [hunk 4]
```diff
@@ -77,24 +77,29 @@ class ManagerError(GravitiException):
...
-    Arguments:
-        is_draft: Whether the status is draft.
-        message: The error message.
 
-    """
+class NoCommitsError(StatusError):
+    """This class defines the exception for illegal operations on dataset with no commit history."""
+
+    def __str__(self) -> str:
+        return f"{super().__str__()}. Please create a draft first"
```
Comment (linjiX) [inline graviti/exception.py:87]: why the message is separated？

**Review by linjiX — APPROVED** [review]

**Pull request merged**